src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -30,6 +33,23 @@ public class UniquePersonList implements Iterable<Person> {
     private final ObservableList<Person> internalList = FXCollections.observableArrayList(extractor);
     private final ObservableList<Person> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
+    private Comparator<Person> comparator = COMPARATOR_UNGROUP_PATIENTS;
+
+    /**
+     * Constructs a UniquePersonList with an added listener to sort list when patients are updated.
+     */
+    public UniquePersonList() {
+        internalList.addListener((ListChangeListener<Person>) c -> {
+            while (c.next()) {
+                if (c.wasUpdated()) {
+                    sort(comparator);
+                }
```

Comment:
I realized in mine I did not have to add this so it might be the same for yours as well.